### ISSUE TITLE
Trinity input not being created?

### ISSUE THREAD

**Issue by nickmiller-iit** [clemgoub/dnaPipeTE#80]
Hello I am having difficulty running dnaPipeTE from a Singularity container.

My organism has a genome of around 1.8Gb

When I start the Singularity container with:

```
singularity shell --bind ./Project:/mnt ./dnapipete.img
```

And then run dnaPipeTE with

```
python3 /opt/dnaPipeTE/dnaPipeTE.py -input /mnt/B_CKDN220061309-1A_HMN35DSX5_L4_1.fq.gz -genome_size 1800000000 -genome_coverage 0.1 -output /mnt/out
```

The run fails, with the initial part of the output as follows:

```
           Let's go !!!

Start time: Wed Mar 29 15:10:46 2023
generating trinity samples...
gz compression detected for /mnt/B_CKDN220061309-1A_HMN35DSX5_L4_1.fq.gz
total number of reads: 434254735
maximum number of reads to sample:  11612902
fastq :  /mnt/B_CKDN220061309-1A_HMN35DSX5_L4_1.fq
sampling 2 samples of max 5806451 reads to reach coverage...
180000009 bases sampled in 1198243 reads
s_B_CKDN220061309-1A_HMN35DSX5_L4_1.fq done.
180000140 bases sampled in 7004665 reads
s_B_CKDN220061309-1A_HMN35DSX5_L4_1.fq done.
['s0_B_CKDN220061309-1A_HMN35DSX5_L4_1.fq.fasta', 's1_B_CKDN220061309-1A_HMN35DSX5_L4_1.fq.fasta']
###################################
### TRINITY to assemble repeats ###
###################################

***** TRINITY iteration 1 *****

Selecting reads for Trinity iteration number 1...
awk: cannot open /mnt/out/Trinity_run0/chrysalis/readsToComponents.out.sort (No such file or directory)
Done
```

Looking at the the contents of the output directory:

```
ls -lFh Project/out/
total 364M
drwxr-sr-x 2 nick nick   10 Mar 29 15:47 Annotation/
-rw-r--r-- 1 nick nick    0 Mar 29 15:47 reads_run0.fasta
-rw-r--r-- 1 nick nick    0 Mar 29 15:47 reads_run1.fasta
-rw-r--r-- 1 nick nick 181M Mar 29 15:41 s0_B_CKDN220061309-1A_HMN35DSX5_L4_1.fq.fasta
-rw-r--r-- 1 nick nick 184M Mar 29 15:47 s1_B_CKDN220061309-1A_HMN35DSX5_L4_1.fq.fasta
-rw-r--r-- 1 nick nick    0 Mar 29 15:47 Trinity.fasta
drwxr-sr-x 2 nick nick   10 Mar 29 15:47 Trinity_run1/
drwxr-sr-x 2 nick nick   10 Mar 29 15:47 Trinity_run2/
```

The Annotation and Trinity_run* directories are empty.

It looks like the input files needed by Trinity are not being created.

Do you have any suggestions?

Thank you

**Comment by clemgoub** [comment]
Hello there!

I think the issue is due to a particularity of the singularity container. The source container is a docker, and it is built such as the default directory is `/opt/dnaPipeTE` which contains all the executables. I found out recently that this was not the case for the singularity container. Thus, I would advise to try to `cd /opt/dnaPipeTE` before running the python3 command.

Your strategy (calling `/opt/dnaPipeTE/dnaPipeTE.py`) works partially because the sampling is made directly from the script `/opt/dnaPipeTE/dnaPipeTE.py` but the call to Trinity is expected to come from `/opt/dnaPipeTE`. 

I noticed that I didn't update the documentation on Github (only [here](https://tehub.org/tutorials/docs/dnaPipeTE)), so I'm doing it now!

Let me know if this does it!

Cheers,

Clément
